snaps/openvino/gpu-support/snap/snapcraft.yaml

```diff
@@ -0,0 +1,145 @@
+      - opengl # GPU device
+      - home
+      - network-bind # needed by jupyter
+      - hugepages-control
+    environment:
+      PYTHONPATH: $SNAP/usr/lib/python3/dist-packages:$SNAP/openvino/usr/python:$SNAP/openvino/usr/python/openvino/tools/ovc:$PYTHONPATH
+      LD_LIBRARY_PATH: $SNAP/openvino/usr/runtime/lib/intel64:$SNAP/openvino/usr/lib/x86_64-linux-gnu:$SNAP/usr/lib/npu-libs-2404:$SNAP/usr/local/lib:$LD_LIBRARY_PATH
+      PATH: $SNAP/openvino/usr/python/openvino/tools/ovc:$PATH
+      OCL_ICD_VENDORS: $SNAP/etc/OpenCL/vendors
+
+parts:
+  openvino-python-deps:
+    # This part will install a bunch of python tools and libraries that
+    # are needed to run and validate the GPU device notebook at:
+    # https://github.com/openvinotoolkit/openvino_notebooks/blob/latest/notebooks/gpu-device
+    source-type: git
+    source: https://github.com/openvinotoolkit/openvino.git
+    plugin: python
+    python-requirements: [src/bindings/python/requirements.txt]
+    python-packages: [ipython, jupyterlab, ipywidgets, opencv-python, tqdm, huggingface_hub, matplotlib>=3.4]
+    stage-packages:
+      - libva2 # openvino gpu plugin dep
+      - libgl1 # opencv-python dep
+      - libglu1-mesa # opencv-python dep
+      - libxcb1 # opencv-python dep
+      - libice6 # opencv-python dep
+      - libsm6 # opencv-python dep
+      - libx11-6 # opencv-python dep
+      - libxext6 # opencv-python-dep
+    stage:
+      - -usr/lib/x86_64-linux-gnu/libGLU.so.1.3.1
+      - -usr/lib/x86_64-linux-gnu/libGLX_mesa.so.0.0.0
+      - -usr/lib/x86_64-linux-gnu/libicuio.so.74.2
+      - -usr/lib/x86_64-linux-gnu/libicutest.so.74.2
+      - -usr/lib/x86_64-linux-gnu/libvulkan.so.1.3.275
+      - -usr/local/lib/libopencl-clang.so
+      - -usr/lib/x86_64-linux-gnu/libOpenGL.so.0.0.0
+      - -usr/lib/x86_64-linux-gnu/libXfixes.so.3.1.0
+      - -usr/lib/x86_64-linux-gnu/libXxf86vm.so.1.0.0
+      - -usr/lib/x86_64-linux-gnu/libicutu.so.74.2
+      - -usr/lib/x86_64-linux-gnu/libxcb-dri2.so.0.0.0
+      - -usr/lib/x86_64-linux-gnu/libxcb-glx.so.0.0.0
+      - -usr/lib/x86_64-linux-gnu/libxcb-present.so.0.0.0
+      - -usr/lib/x86_64-linux-gnu/libxcb-randr.so.0.1.0
+      - -usr/lib/x86_64-linux-gnu/libxcb-shm.so.0.0.0
+      - -usr/lib/x86_64-linux-gnu/libxcb-sync.so.1.0.0
+      - -usr/lib/x86_64-linux-gnu/libxcb-xfixes.so.0.0.0
+      - -usr/lib/x86_64-linux-gnu/libxshmfence.so.1.0.0
+      - -usr/local/lib/libopencl-clang.so
+      - -usr/local/lib/libopencl-clang.so.14
+      - -usr/lib/x86_64-linux-gnu/libicui18n.so.74.2
+  opencl-driver:
+    # Includes all the compute runtime and OpenCL bits needed for Intel GPU support
+    plugin: nil
+    build-packages:
+      - wget
+    override-build: |
+      mkdir -p neo
+      cd neo
+      # Install Intel graphics compiler and compute runtime
+      wget https://github.com/intel/intel-graphics-compiler/releases/download/igc-1.0.17791.9/intel-igc-core_1.0.17791.9_amd64.deb
```

Comment:
curious about how you end up with this list of dependencies : `igc`, `level-zero`, `intel-opencl-icd`, i assume from some public documentation so if you can paste the link here, i will be grateful

Reply:
Good call - I'll include a link shortly.